thoth/storages/sync.py

```diff
@@ -61,8 +63,17 @@ def sync_adviser_documents(
         adviser_store = AdvisersResultsStore()
         adviser_store.connect()
 
+    if not document_ids:
+        document_listing = list(adviser_store.get_document_listing())
+
+    if RANDOMIZE_LISTING:
+        if document_ids:
+            random.shuffle(document_ids)
+        else:
+            random.shuffle(document_listing)
+
     processed, synced, skipped, failed = 0, 0, 0, 0
-    for document_id in document_ids or adviser_store.get_document_listing():
+    for document_id in document_ids or document_listing:
```

Comment:
The snippet above might be prone to programmatic errors as the if-else is basically done twice. It might be a good idea to be explicit which listing is actually used just once:

```suggestion
    if _RANDOMIZE_LISTING:
        listing = random.shuffle(document_ids or document_listing)
    for document_id in listing:
```